### DISCUSSION TITLE
Mau tanya dan minta saran

### DISCUSSION THREAD

**Discussion by dimaspamungkasmursidi** [deaafrizal/forum#58] [The Lounge]
Halo guys mau tanya..
Sebelumnya gua ceritain detailnya dulu, jadi insya allah gua mau fokus belajar frontend js dgn konsisten di tahun ini. Alhamdulillah HTML dan CSS udah bisa, nah masalah di javascript nya ini, sebenernya udh belajar js dari 2022 ya kira-kira 1,5 tahun, tapi ga serius belajarnya, nah pas agak bener belajarnya pas tahun lalu karna belajar web dev di hacktiv, tp ya gitu masih kureng serius, jadi hasilnya gua paham code js nya, tapi klo misal disuru buat project sendiri ga bisa tanpa bantuan gugel, contohnya gini, misal gua ikutin tutor js bang dea yang saklar lampu itu atau tutor js bang dea yg lain, gua paham codenya itu buat apa, kaya onclick itu buat apa, if else buat apa, lopping, map, dll itu paham, tapi klo misal disuru buat project sendiri tu ga bisa, jadi harus pake bantuan dulu baru bisa entah itu dari stack overflow, chat gpt, atau yt.

**_Nah jadi klo kaya gitu gmn yaa guys? apa udh boleh pindah ke ke react? atau harus balik lg ke js? atau harus gmn?_**

Sebenernya sekarang gua lg belajar react, dan udah bisa yg dasar-dasar kaya compopnent, props, react router, conditional render, rendering list, yg simpel-simpel gitu pokonya, usestate dan seterusnya belum. Sekarang ini lg kepikiran bgt itu, kaya bingung gitu harus gmn alur yg benernya krn skrrg otw semester 8 tapi baru bisa yang itu-itu aja, takut ga bisa dapet kerja, mana skrg lg tech winter :) hehe..

btw klo ada yg lagi belajar kaya gua atau lg mau bikin project buat belajar, boleh dong belajar bareng hehe, ngobrol-ngobrol, sekalian belajar kerja dalam tim..

**Comment by ikhlasdansantai** [comment]
**point pertama**, bantuan dari google, dokumentasi dan semacamnya itu wajar dan sah-sah saja, karena kita ini manusia bro, bukan robot, pasti ada lupa nya, makanya kita butuh **throwback** alias mengulas kembali apa yang udah kita pelajari sebelumnya.

**point kedua**, apakah sudah waktunya untuk pake react?, karena lu udah belajar js sebelumnya, apalagi udah 1tahun lebih, saran gw pelajari aja react nya, karena react itu basicnya ya javascript javascript juga, jadi bisa sambil dipelajari javascript nya dari react. atau misalkan kesusahan karena nemu konsep atau materi baru di react yang belum pernah ditemui sebelumnya, ya kembali dulu ke javascript dasar.

sedikit tambahan materi/konsep yang perlu dipelajari klo mau pake react
basics: 
1. scoping
2. callbacks
3. passing functions
4. reference vs value
5. == vs ===
advance:
1. short circuiting
2. advanced array methods
3. immutability
4. asynchronous
5. concepts of modules
6. concepts of es6 ( i.e. spread operator)

**Reply by dimaspamungkasmursidi** [reply]
makasih bgt bang atas jawabannya.. Barakallahu fiik..

**Reply by ikhlasdansantai** [reply]
siap bg, karena lu ini fokus nya ke FE, gw saranin buat latihan slicing dulu di
- https://codedesign.dev/
- https://www.frontendmentor.io/challenges  

ya minimal buat 1 web utuh aja cukup atau lebih juga lebih baik.

untuk **step selanjutnya**, bisa bikin web dengan **consume api**, bang dea pernah buat tuh, yang ini
- https://youtu.be/QV4qKaeEf9c?si=RTDMPzynUcxvYdXT
bisa ikutin aja videonya.

**Reply by dimaspamungkasmursidi** [reply]
Alhamdulillah bang klo slicing udh krn project bootcampnya kaya gitu + rangkap jadi ui/ux juga dulu pas bootcamp.
nah paling consume api nih harus banyak belajar lagi.. thank u bgt bang..
